targets/TARGET_STM/TARGET_STM32L5/system_clock.c

```diff
@@ -177,17 +177,6 @@ uint8_t SetSysClock_PLL_MSI(void)
         return 0; // FAIL
     }
 
-    // Default STDIO is LPUART1
-    RCC_PeriphCLKInitTypeDef PeriphClkInitStruct = {0};
-#if DEVICE_TRNG
-    PeriphClkInitStruct.PeriphClockSelection = RCC_PERIPHCLK_LPUART1 | RCC_PERIPHCLK_RNG;
```

Comment:
You are also removing RCC_PERIPHCLK_RNG configuration...
Maybe this has impact in 
https://github.com/ARMmbed/mbed-os/blob/master/targets/TARGET_STM/trng_api.c#L84-L85